internal/cmd/agent.go

```diff
@@ -35,34 +34,43 @@ func startCmd() *cobra.Command {
 	var (
 		token    string
 		coderURL string
+		logFile  string
 	)
 	cmd := &cobra.Command{
-		Use:   "start --coder-url=[coder_url] --token=[token]",
+		Use:   "start --coder-url=<coder_url> --token=<token> --log-file=<path>",
 		Short: "starts the coder agent",
 		Long:  "starts the coder agent",
 		Example: `# start the agent and use CODER_URL and CODER_AGENT_TOKEN env vars
-
 coder agent start
 
 # start the agent and connect with a specified url and agent token
-
 coder agent start --coder-url https://my-coder.com --token xxxx-xxxx
+
+# start the agent and write a copy of the log to /tmp/coder-agent.log
+# if the file already exists, it will be truncated
+coder agent start --log-file=/tmp/coder-agent.log
 `,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
 			sinks := []slog.Sink{
 				sloghuman.Sink(os.Stderr),
 			}
 
-			file, err := os.OpenFile(filepath.Join(os.TempDir(), "coder-agent.log"), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
-			if err == nil && file != nil {
+			// Optional log file path to write
+			if logFile != "" {
+				// Truncate the file if it already exists
+				file, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+				if err != nil {
+					// If an error occurs, it should be fatal, because we asked to write
+					// a log to a path but cannot write it for some reason
+					return xerrors.Errorf("open log file: %w", err)
```

Comment:
@coadler since the log file is optional now, if it's specified and cannot be written, then it's a fatal error

Reply:
@coadler @deansheather Actually, I'm not sure if making this fatal is a good idea, since it'd prevent workspaces from starting in that case :thinking: 

What do y'all think?

Reply:
probably not fatal then